src/pages/Scan/index.tsx

```diff
@@ -0,0 +1,104 @@
+import { useEffect, useRef, useState } from 'react';
+import { Link } from 'react-router-dom';
+import jsQR from 'jsqr';
+import styles from './index.module.scss';
+import cn from 'classnames';
+
+const Scan = () => {
+  const videoRef = useRef(null);
+  const canvasRef = useRef(null);
+  const cameraScannerIntervalRef = useRef(null);
+  const [cameraStream, setCameraStream] = useState<MediaStream | null>(null);
+  const [linkQrCode, setLinkQrCode] = useState('');
+
+  useEffect(() => {
+    if (navigator.mediaDevices && navigator.mediaDevices.getUserMedia) {
+      navigator.mediaDevices
+        .getUserMedia({
+          video: {
+            facingMode: 'environment',
+          },
+        })
+        .then(stream => {
+          setCameraStream(stream);
+          const cameraElement = videoRef.current;
+          cameraElement.srcObject = stream;
+          cameraElement.onloadedmetadata = () => {
+            cameraElement.play();
+          };
+        })
+        .catch(error => console.error('Error accessing camera', error));
+    } else {
+      alert('getUserMedia() is not supported by your browser');
+    }
+  }, []);
+
+  useEffect(() => {
+    const cameraElement = videoRef.current;
+    const canvasElement = canvasRef.current;
+    if (cameraElement && canvasElement) {
+      cameraElement.onloadeddata = () => {
+        cameraScannerIntervalRef.current = setInterval(() => {
+          if (cameraElement.videoWidth > 0 && cameraElement.videoHeight > 0) {
+            const canvas = canvasElement;
+            canvas.width = cameraElement.videoWidth;
+            canvas.height = cameraElement.videoHeight;
+            const ctx = canvas.getContext('2d', { willReadFrequently: true });
+            ctx.drawImage(cameraElement, 0, 0, canvas.width, canvas.height);
+            const imageData = ctx.getImageData(0, 0, canvas.width, canvas.height);
+            const codeData = jsQR(imageData.data, imageData.width, imageData.height);
+
+            if (codeData) {
+              setLinkQrCode(codeData.data);
+              if (videoRef.current) {
+                videoRef.current.pause();
+              }
+            }
+          }
+        }, 1000);
+      };
+    }
+
+    const stopCamera = (cameraStream: MediaStream | null) => {
+      if (cameraStream) {
+        cameraStream.getTracks().forEach(track => track.stop());
+      }
+    };
+
+    return () => {
+      if (cameraScannerIntervalRef.current) {
+        clearInterval(cameraScannerIntervalRef.current);
+      }
+      stopCamera(cameraStream);
+    };
+  }, [cameraStream]);
+
+  const handleLinkClick = (event: React.MouseEvent<HTMLAnchorElement>) => {
+    event.preventDefault();
+    if (linkQrCode) {
+      window.location.href = linkQrCode;
```

Comment:
@satanmourner it should not go any link that scanned link must contain some string and at last use Navigate to redirect to the path link pattern would be `https://wallet.socious.io/connect?_oob=<STRING>&callback=<STRING>` 
please extract path and queries then if match navigate to '/connect` with queries if not QR is not valid